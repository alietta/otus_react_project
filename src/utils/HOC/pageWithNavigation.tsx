import React from "react";
import { Redirect } from "react-router-dom";
import { DarkMode, Container, useTheme } from "sancho";
import { pageStyle } from "./pageStyle";
import { Headbar } from "components/Headbar";
import { logout } from "@/api/auth";

export const pageWithNavigation = <Props extends object>(
  Component: React.ComponentType<Props>
) => (props: Props) => {
  const theme = useTheme();
  console.warn(props)
  const onLogout = () => {
    (async () => {
      await logout();
      debugger
      props.setStore({name: '', isAuth: false})
    })();
  };
  return (
    <div
      css={{
        width: "100vw",
        height: "100vh",
        background: theme.colors.background.default,
      }}
    >
      <Headbar name={props.store.name} onLogout={onLogout} />
      <Container css={{ maxWidth: 1024, paddingTop: 50 }}>
        <Component {...props} />
      </Container>
      {/* {needRedirect && <Redirect to={{ pathname: "/login" }} />} */}
    </div>
  );
};
