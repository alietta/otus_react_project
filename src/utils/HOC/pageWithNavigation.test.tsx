import React from "react";
import { mount } from "enzyme";
import { pageWithNavigation } from "./pageWithNavigation";

describe("HOC pageWithNavigation", () => {
  interface ComponentProps {
    store: {
      name: string;
    }
  }
  const Component: React.FC<ComponentProps> = ({ store: {name }}) => <h1>{name}</h1>;
  const WrappedComponent = pageWithNavigation(Component);
  it("render", () => {
    const element = mount(<WrappedComponent store={{name: "Anny"}}/>);
    expect(element.html()).toMatchSnapshot();
  });
});
