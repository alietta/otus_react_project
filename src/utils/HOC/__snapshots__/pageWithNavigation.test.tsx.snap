// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC pageWithNavigation render 1`] = `"<div class=\\"css-1xfup8k\\"><div class=\\"css-94ghjt-Headbar\\"><div class=\\"css-hkw5ma-Headbar\\"><span class=\\"css-69rko3-Headbar\\">Anny</span><div class=\\"logout-icon css-1vhmik5-Headbar\\"><svg viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"#212529\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" height=\\"20px\\" width=\\"20px\\"><path d=\\"M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4\\"></path><polyline points=\\"16 17 21 12 16 7\\"></polyline><line x1=\\"21\\" y1=\\"12\\" x2=\\"9\\" y2=\\"12\\"></line></svg></div></div></div><div class=\\"css-hr2t1h\\"><h1>Anny</h1></div></div>"`;
